### ISSUE TITLE
Padding with 0 in mel2samp is harmful

### ISSUE THREAD

**Issue by nshmyrev** [NVIDIA/waveglow#95]
This code in mel2samp.py pads the small audio with zeros, causing loss instability:

```
        # Take segment
        if audio.size(0) >= self.segment_length:
            max_audio_start = audio.size(0) - self.segment_length
            audio_start = random.randint(0, max_audio_start)
            audio = audio[audio_start:audio_start+self.segment_length]
        else:
            audio = torch.nn.functional.pad(audio, (0, self.segment_length - audio.size(0)), 'constant').data

```

it is not relevant for the default setup, since ljspeech samples are longer than 1 second and the segment_length is less than 1 second, but in different datsets it might cause serious issues. Its better to skip such audio files probably or pad with silent white noise instead.

**Comment by jiqizaisikao** [comment]
why?

**Comment by rafaelvalle** [comment]
Closing due to inactivity.

**Comment by rafaelvalle** [comment]
@nshmyrev did you confirm that not padding with zeros fixes the issue?

**Comment by nshmyrev** [comment]
Yes

**Comment by rafaelvalle** [comment]
great, thank you!
